contracts/HookedTokenManager.sol

```diff
@@ -240,6 +245,30 @@ contract HookedTokenManager is ITokenController, IForwarder, AragonApp {
         emit RevokeVesting(_holder, _vestingId, nonVested);
     }
 
+    /**
+    * @notice Wrap @tokenAmount(self.wrappableToken(): address, _amount, false) to receive @tokenAmount(self.token(): address, _amount, false)
+    * @param _amount Amount of tokens to wrap
+    */
+    function wrap(uint256 _amount) public {
+        require(wrappableToken != address(0), ERROR_NO_WRAPPABLE_TOKEN);
+        require(msg.sender != address(this), ERROR_MINT_RECEIVER_IS_TM);
+
+        wrappableToken.safeTransferFrom(msg.sender, address(this), _amount);
+        _mint(msg.sender, _amount);
+    }
+
+    /**
+    * @notice Unwrap @tokenAmount(self.token(): address, _amount, false) to receive @tokenAmount(self.wrappableToken(): address, _amount, false)
+    * @param _amount Amount of tokens to unwrap
+    */
+    function unwrap(uint256 _amount) public {
+        require(wrappableToken != address(0), ERROR_NO_WRAPPABLE_TOKEN);
+        require(msg.sender != address(this), ERROR_MINT_RECEIVER_IS_TM);
+
+        wrappableToken.safeTransfer(msg.sender, _amount);
+        _burn(msg.sender, _amount);
```

Comment:
Wonder if it's a better security practice to call the `_burn` method before we do the token transfer here?